core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -86,7 +86,10 @@ private[spark] object TaskDescription {
     dataOut.writeInt(taskDescription.properties.size())
     taskDescription.properties.asScala.foreach { case (key, value) =>
       dataOut.writeUTF(key)
-      dataOut.writeUTF(value)
+      // SPARK-19796 -- writeUTF doesn't work for long strings, which can happen for property values
+      val bytes = value.getBytes("utf-8")
+      dataOut.writeInt(bytes.length)
+      dataOut.write(bytes)
```

Comment:
Can you just use .writeObject? I just read the code for this and it looks like that calls writeUTF() under the hood when the string is short enough, and otherwise calls writeUTF a few times.  It seems better to use that built-in than to try to do it ourselves.